### PR TITLE
Update finetuning regressor loss computation to utilize the new target transformation function.

### PR DESCRIPTION
**Pull request by anuragg1209**
## Issue
Please link the corresponding GitHub issue. If an issue does not already exist,
please open one to describe the bug or feature request before creating a pull request.

This allows us to discuss the proposal and helps avoid unnecessary work.

## Motivation and Context

- Changed dataset from California Housing to Diamonds (OpenML 42225) in finetune_regressor.py.
- Introduced a new function to map targets into the estimator's prediction space in finetuned_regressor.py.
- Updated regression loss computation to utilize the new target transformation function.
- Added tests to verify the target transformation functionality.

## Public API Changes

-   [ ] No Public API changes
-   [ ] Yes, Public API changes (Details below)

---

## How Has This Been Tested?

Locally tested end to end. 

## Checklist

-   [x] The changes have been tested locally.
-   [x] Documentation has been updated (if the public API or usage changes).
-   [x] A changelog entry has been added (see `changelog/README.md`), or "no changelog needed" label requested.
-   [x] The code follows the project's style guidelines.
-   [x] I have considered the impact of these changes on the public API.

---

### CODE REVIEW

File: src/tabpfn/finetuning/finetuned_regressor.py [hunk 1]
```diff
@@ -33,9 +33,24 @@
...
+) -> torch.Tensor:
+    """Map z-scored targets into the space the estimator predicts in.
+
+    Members with a target transform (the default ensemble's "safepower") saw
+    transformed context targets, so their loss must use transformed targets too.
+    """
+    if config.target_transform is None:
+        return y_Q
+    y_np = config.target_transform.transform(y_Q.cpu().numpy().reshape(-1, 1))
+    return torch.as_tensor(y_np.ravel(), dtype=y_Q.dtype, device=y_Q.device)
```
[the remark this answers is not in the document]
Comment (anuragg1209) [reply src/tabpfn/finetuning/finetuned_regressor.py:51]: Not reachable: the regressor's fit validation (`ensure_compatible_fit_inputs_sklearn` with `ensure_y_numeric=True`) raises "Input y contains NaN" for any NaN target, so no NaN can reach this helper. 

The NaN mask in `_compute_regression_loss` is part of the general bar-distribution loss contract, not a path fine-tuning inputs can hit.

File: src/tabpfn/finetuning/finetuned_regressor.py [hunk 2]
```diff
@@ -390,9 +405,9 @@ def _forward_with_loss(self, batch: RegressorBatch) -> torch.Tensor:  # type: ig
         # permute to shape (B, E, Q, L) then reshape to (B*E, Q, L)
         logits_BQL = logits_QBEL.permute(1, 2, 0, 3).reshape(B * E, Q, L)
 
-        targets_BQ = y_query_batch.repeat(B * self._local_n_estimators_, 1).to(
-            self.device
-        )
+        targets_BQ = torch.stack(
+            [_targets_in_estimator_space(y_query_batch[0], c[0]) for c in batch.configs]
```
[the remark this answers is not in the document]
Comment (anuragg1209) [reply src/tabpfn/finetuning/finetuned_regressor.py:409]: Fixed in latest commit. RegressorBatch/ClassifierBatch now store the fitted configs returned by the ensemble members instead of the originals. Verified with n_preprocessing_jobs=2 that the target transforms in the batch are fitted.

File: src/tabpfn/finetuning/finetuned_regressor.py (hunk 1 above)
Comment (bejaeger) [inline src/tabpfn/finetuning/finetuned_regressor.py:51]: cool, can you mention that fix in the changelog as well?

**Review by bejaeger — DISMISSED** [review]
LGTM! just one comment.
